### DISCUSSION TITLE
graphql-http vulnerable to CSRF attack

### DISCUSSION THREAD

**Discussion by fr4nco1s** [graphql/graphql-http#81] [Q&A]
Hi,

graphql-http seems to be vulnerable to [CSRF attack](https://blog.doyensec.com/2021/05/20/graphql-csrf.html).

Is it planed to implement an anti-CSRF system, like tokens management on GraphQL endpoints ?

**Comment by enisdenjo** [comment]
Server-side of graphql-http is just a handler purely implementing the [GraphQL over HTTP spec](https://graphql.github.io/graphql-over-http/) (see [disclaimer in readme](https://github.com/graphql/graphql-http#only-graphql-over-http)).

All other additions that go out of the GraphQL transport scope are to be implemented user-land - exactly why graphql-http is a handler, and not a server. Anti-CSRF systems, CORS, encrypted cookies and token management is not something the core of this library intends to tackle.
